### PR TITLE
Adding structured logging example

### PR DESCRIPTION
**Pull request by jim-sheldon**
Includes logging levels

### CODE REVIEW

File: log.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  )
+
+  # Convert log object to JSON
+  log_json <- toJSON(log, pretty = TRUE)
+
+  # Print JSON log
+  cat(log_json, "\n")
+}
+"""
+
```
Comment (calremmel) [inline log.py:28]: consider: put all this under `if __name__ == "__main__":`?

**Review by calremmel — APPROVED** [review]
Wow you're fast! I ran it and it works great, thanks for this!

Output:

```
{
  "timestamp": ["2024-05-17 09:22:54"],
  "level": ["INFO"],
  "message": ["Hello!"]
}
{
  "timestamp": ["2024-05-17 09:22:54"],
  "level": ["CRITICAL"],
  "message": ["Uh oh!"]
}
```


I do have one change request: could you add a `requirements.txt` and note the Python version used? I used 3.12 and it ran fine.

I know that this whole exercise is throwaway-ish, so approving so you have the flexibility to do this in your other PR if you would prefer.

**Comment by jim-sheldon** [comment]
> I do have one change request: could you add a `requirements.txt` and note the Python version used? I used 3.12 and it ran fine.

Yeah, I thought about this a little late, and added a `pyproject.toml` file in #3

File: log.py (hunk 1 above)
Comment (jim-sheldon) [reply log.py:28]: eh, it's all spikes

**Pull request merged**